vispy/util/tests/test_event_emitter.py

```diff
@@ -663,4 +663,81 @@ def assert_state(a, b):
     assert_state(True, True)
 
 
+def test_emitter_reentrance_allowed_when_blocked1():
+    # Minimal re-entrance example
+
+    e = EventEmitter(source=None, type='test')
+    count = 0
+
+    @e.connect
+    def foo(x):
+        nonlocal count
+        count += 1
+        with e.blocker():
+            e()
+
+    e()
+    assert count == 1
+
+
+def test_emitter_reentrance_allowed_when_blocked2():
+    # More realistic re-entrance example
+    # The real world case for this might be a longer chain of events
+    # where event1's callback knows that what it's doing will trigger
+    # the same event so it blocks it. So event1 -> foo -> event2
+    # -> bar -> event1 -> (ignored bc blocked).
+
+    e1 = EventEmitter(source=None, type='test1')
+    e2 = EventEmitter(source=None, type='test2')
+    count = 0
+
+    @e1.connect
+    def foo(x):
+        nonlocal count
+        count += 1
+        with e1.blocker():
+            e2()
+
+    @e2.connect
+    def bar(x):
+        nonlocal count
+        count += 10
+        e1()
+
+    e1()
+    assert count == 11
+
+
+def test_emitter_reentrance_allowed_when_blocked3():
+    # Like the previous, but blocking callbacks instead of the event itself.
+    # Allows more fine-grained control. To some extent anyway - all callbacks
+    # of the event must be blocked to prevent raising the emitter loop error.
```

Comment:
I don't fully understand the use-case for this, TBH. but I also don't see any harm.

Reply:
I'm sure a use case for this would be something like resize event that triggers some transform updates that do some other kind of resize-like event...I don't know.

Anyway, I don't understand the final assertion value here.

Count starts at 0.

```
e1() -> foo() (+ 1) -> e2() ->
    bar() (+ 10) -> e1() -> blocked callback
    eggs() (+ 100) -> e1() -> blocked callback
```

So 1 + 10 + 100 => 111, right? Where does the 2000 come from?

Reply:
:man_facepalming: Nevermind, I was looking at the original unfixed commit.

Reply:
Yeah sorry, I added something to the test which I removed later, but forgot to update the check.